tests/Security/Voter/HandleCommandVoterTest.php

```diff
@@ -60,21 +47,34 @@ public function provideTestVoteData()
     {
         return [
             // Testcase: default access is false
-            ['handle', new FakeCommand, true, [], VoterInterface::ACCESS_DENIED],
+            ['handle', new FakeCommand, [new Role('ROLE_ADMIN')], [], VoterInterface::ACCESS_DENIED],
+
             // Testcase: abstain when not handling a command, but using the handle attribute
-            ['handle', null, true, [], VoterInterface::ACCESS_ABSTAIN],
+            ['handle', null, [new Role('ROLE_ADMIN')], [], VoterInterface::ACCESS_ABSTAIN],
+
             // Testcase: abstain when not handling a command and not using the handle attribute
-            ['create', null, true, [], VoterInterface::ACCESS_ABSTAIN],
+            ['create', null, [new Role('ROLE_ADMIN')], [], VoterInterface::ACCESS_ABSTAIN],
+
             // Testcase: abstain when not handling a command
-            ['create', new FakeCommand, true, [], VoterInterface::ACCESS_ABSTAIN],
-            // Testcase: default is unrelated to decision manager
-            ['handle', new FakeCommand, false, [], VoterInterface::ACCESS_DENIED],
-            // Testcase: deny access if decision manager returns false
-            ['handle', new FakeCommand, false, [FakeCommand::class => ['ROLE_USER']], VoterInterface::ACCESS_DENIED],
-            // Testcase: grant access if decision manager returns true and the command is in the mapping
-            ['handle', new FakeCommand, true, [FakeCommand::class => ['ROLE_USER']], VoterInterface::ACCESS_GRANTED],
+            ['create', new FakeCommand, [new Role('ROLE_ADMIN')], [FakeCommand::class => ['ROLE_ADMIN']], VoterInterface::ACCESS_ABSTAIN],
+
+            // Testcase: default is unrelated to roles
+            ['handle', new FakeCommand, [new Role('ROLE_ADMIN')], [], VoterInterface::ACCESS_DENIED],
+
+            // Testcase: deny access if incorrect role
+            ['handle', new FakeCommand, [new Role('ROLE_ADMIN')], [FakeCommand::class => ['ROLE_USER']], VoterInterface::ACCESS_DENIED],
+
+            // Testcase: grant access if the user has the configure role
+            ['handle', new FakeCommand, [new Role('ROLE_USER')], [FakeCommand::class => ['ROLE_USER']], VoterInterface::ACCESS_GRANTED],
+
+            // Testcase: grant access if the user has one of the configure roles
+            ['handle', new FakeCommand, [new Role('ROLE_USER')], [FakeCommand::class => ['ROLE_USER', 'ROLE_TWO']], VoterInterface::ACCESS_GRANTED],
+
+            // Testcase: grant access if the user has one of the configure roles, but also another role
```

Comment:
By the way, you can put these descriptions as the key to the parameters array and PHPUnit will then show that name when the test fails. :)

e.g. 

```
'2+2 is 4' => [2, 2, 4]
```